_posts/en/newsletters/2021-03-24-newsletter.md

```diff
@@ -0,0 +1,299 @@
+    currently usable on mainnet for anyone who wants to experiment with
+    it.
+
+- **Discussion of quantum computer attacks on taproot:**
+  the original Bitcoin software provided two ways to receive bitcoin:
+
+  - *Pay-to-Public-Key (P2PK)* implemented the simple and clear method
+    described in the [original Bitcoin paper][] of receiving bitcoins to
+    a public key and allowing those coins to be spent by a signature.  The
+    Bitcoin software used this by default when the public key material
+    could be handled entirely by software.
+
+  - *Pay-to-Public-Key-Hash (P2PKH)* added a layer of indirection,
+    receiving bitcoins to a hash digest that committed to the public key
+    to be used.  To spend the coins, the public key would need to be
+    published alongside the signature, making the 20 bytes dedicated to
+    the hash digest an overhead cost.  This was used by default when the
+    payment information might need to be handled by a person, e.g. an
+    address that could be copied and pasted.
+
+  Nakamoto never described why he implemented both methods, but it's
+  widely believed that he added the hash indirection in order to make
+  Bitcoin addresses smaller so that they could be communicated more
+  easily.  Public keys in the original Bitcoin implementation were 65
+  bytes, but address hashes were only 20 bytes.
+
+  In the decade since, there have been a number of developments.  To
+  make certain multisig protocols [simple and secure by default][bech32
+  address security], it was determined that scripts for multiparty
+  protocols should probably use a 32-byte commitment.  Bitcoin developers also
+  learned about previously known techniques that could compress a public
+  key down to 33 bytes---later [optimized][news48 oddness] to 32 bytes.
+  Finally, [taproot's primary innovation][news27 taproot] showed that a
+  32-byte public key could commit to a script with [security][news87
+  fournier proof] similar to that of a 32-byte hash.  All of this means
+  that it no longer changes the amount of address data people have to
+  communicate whether they use a hash or a public key---it's 32 bytes
+  either way if they want a universally applicable address format.
+  However, directly using public keys still eliminates the extra
+  bandwidth and storage resulting from hash indirection.  If every
+  payment went to a public key instead of a 32-byte hash, it  would save
+  about 13 gigabytes of block chain space per year.  <!-- 32 bytes *
+  8000 inputs per block * 52416 blocks per year --> The [BIP341][]
+  specification of taproot describes space savings as the reason it
+  accepts payments to public keys in the P2PK style instead of hashes in
+  the P2PKH style.
+
+  But P2PKH hash indirection does have one advantage: it can hide keys
+  from public view until they're needed to authorize a spend.  This
+  means an adversary who has the ability to compromise a public key
```

Comment:
maybe "ability to compromise a public key" -> "ability to calculate a private key from the public key" ?